test/support/utils_test_base.h

```diff
@@ -319,10 +319,12 @@ struct test_base
 
 namespace
 {
+[[maybe_unused]]
```

Comment:
This fix looks super weird. The right thing to do it use this function properly.

Let's just make it 

template <typename TTestDataTransfer>
void retrieve_data(TTestDataTransfer& helper)
{
    helper.retrieve_data();
}

The same for the function below

Reply:
Please explain.